src/flownet/network_model/_generate_connections.py

```diff
@@ -94,9 +94,46 @@ def _remove_recorded_connections(
     return conn_matrix
 
 
+def _split_additional_flow_nodes(
+    configuration: Any, concave_hull_bounding_boxes: np.ndarray
+) -> List:
+    """
+    This function takes splits the additional_flow_nodes defined in the config over the layers. The division is based on the size (volume) of the boundingbox of the layer.
+
+    Args:
+        configuration: FlowNet configuration yaml as dictionary
+        concave_hull_bounding_boxes: List of boundingbox per layer, i.e., numpy array with x, y, z min/max
+            boundingboxes for each grid block
+
+    Returns:
+        List of additional notes per layer (len of list same as number of layers)
+
+    """
+    total_additional_nodes = configuration.flownet.additional_flow_nodes
+    volumes = []
+    for xyz in concave_hull_bounding_boxes:
+        volume = sum(
+            [
+                (
+                    (xyz[i, 1] - xyz[i, 0])
+                    * (xyz[i, 3] - xyz[i, 2])
+                    * (xyz[i, 5] - xyz[i, 4])
+                )
+                for i in range(0, xyz.shape[0])
+            ]
+        )
+        volumes.append(volume)
+
+    fractions = [v / sum(volumes) for v in volumes]
+    sep_add_flownodes = [round(frac * total_additional_nodes) for frac in fractions]
+
+    return sep_add_flownodes
+
+
 def _generate_connections(
     df_coordinates: pd.DataFrame,
     configuration: Any,
+    additional_flownodes: int,
```

Comment:
Here you add the `additional_flownodes`, because you have already split them before going into this function.

However, as you also pass the configuration and the `concave_hull_bounding_boxes list` you already have all required info and you could do the splitting of additional nodes _inside_ this function, isn't it?

Oh by the way, maybe call this variable `additional_flow_nodes` to be aligned with underscores in earlier naming...